plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -165,25 +163,53 @@ public OpenShiftConnector(DockerConnectorConfiguration connectorConfiguration,
      */
     @Override
     public ContainerCreated createContainer(CreateContainerParams createContainerParams) throws IOException {
-        String containerName = getNormalizedContainerName(createContainerParams);
+        String containerName = KubernetesStringUtils.getContainerName(createContainerParams.getContainerName());
         String workspaceID = getCheWorkspaceId(createContainerParams);
+
         // Generate workspaceID if CHE_WORKSPACE_ID env var does not exist
         workspaceID = workspaceID.isEmpty() ? generateWorkspaceID() : workspaceID;
-        String imageName = createContainerParams.getContainerConfig().getImage();
+
+        // imageForDocker is the docker version of the image repository. It's needed for other
+        // OpenShiftConnector API methods, but is not acceptable as an OpenShift name
+        String imageForDocker = createContainerParams.getContainerConfig().getImage();
+        // imageStreamTagName is imageForDocker converted into a form that can be used
+        // in OpenShift
+        String imageStreamTagName = KubernetesStringUtils.getTagNameFromRepoString(imageForDocker);
+
+        // imageStreamTagName is not enough to fill out a pull spec; it is only the tag, so we
+        // have to get the ImageStreamTag from the tag, and then get the full ImageStreamTag name
+        // from that tag. This works because the tags used in Che are unique.
+        ImageStreamTag imageStreamTag = getImageStreamTagFromRepo(imageStreamTagName);
+        String imageStreamTagPullSpec = imageStreamTag.getMetadata().getName();
+
+        // Next we need to get the address of the registry where the ImageStreamTag is stored
+        String imageStreamName = imageStreamTagPullSpec.split(":")[0];
```

Comment:
Agreed. I didn't do it before because OpenShift *should* complain that an ImageStreamTag name does not contain ":", but I'm not sure. Fixed.